lib/events/dynamoevents/dynamoevents.go

```diff
@@ -275,6 +321,78 @@ const (
 	tableStatusOK
 )
 
+var migrateRFD24FinishedKey = backend.Key("internal", "dynamoevents", "rfd_24_migration_complete")
+
+// migrateRFD24 checks if any migration actions need to be performed
+// as specified in RFD 24 and applies them as needed.
+func (l *Log) migrateRFD24(ctx context.Context, dataBackend backend.Backend) error {
+	hasIndexV1, err := l.indexExists(l.Tablename, indexTimeSearch)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	migrateIndices := true
+	migrateEvents := true
+
+	// Table is already up to date.
+	if !hasIndexV1 {
+		migrateIndices = false
+
+		_, err := dataBackend.Get(ctx, migrateRFD24FinishedKey)
+		if err == nil {
+			migrateEvents = false
+		} else if !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	if migrateIndices {
+		log.Info("DynamoDB Index still in old format. Starting migration...")
+
+		hasIndexV2, err := l.indexExists(l.Tablename, indexTimeSearchV2)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		log.Info("Creating new DynamoDB index...")
+		if !hasIndexV2 {
+			err = l.createV2GSI()
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		}
+
+		log.Info("Removing old DynamoDB index")
+		err = l.removeV1GSI()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	if migrateEvents {
+		go func() {
+			log.Info("Starting event migration to v6.2 format")
+			err := l.migrateDateAttribute(ctx)
+			if err != nil {
+				log.WithError(err).Error("Encountered error migrating events to v6.2 format")
+				return
+			}
+
+			item := backend.Item{
+				Key:   migrateRFD24FinishedKey,
+				Value: make([]byte, 0),
+			}
+
+			_, err = dataBackend.Put(ctx, item)
+			if err != nil {
+				log.WithError(err).Error("Migrated all events to v6.2 format successfully but failed to write flag to backend.")
+			}
+		}()
```

Comment:
If event migration fails, it won't be attempted again until teleport restarts.  This isn't ideal.  Instead, event migration should continue to be attempted until either it succeeds, or another auth server successfully performs the migration (this should be done on a fairly long and jittered interval).

Reply:
Added migration retries on a jittered interval.